remote/remote.go

```diff
@@ -49,6 +50,24 @@ type Remote interface {
 
 	// GetHook gets the hook from the http Request.
 	GetHook(r *http.Request) (*model.Hook, error)
+
+	// GetStatusHook gets the status hook from the http Request.
+	GetStatusHook(r *http.Request) (*model.StatusHook, error)
```

Comment:
Let's combine these into a single `GetHook` function like this:

```
type Hook struct {
  Kind string
  IssueComment IssueCommentHook
  Status       StatusHook
}
```

then we can do the following:

```
switch hook.Kind {
case "status":
...
case "issue_comment:
...
}
```